apps/api/src/app/modules/user/user.service.ts

```diff
@@ -76,4 +78,52 @@ export class UserService extends CrudService<User> {
     });
   }
 
+  async updateProfile(entity: any): Promise<User> {
+
+    const currentUser = RequestContext.currentUser();
+    const userId = currentUser?.id;
+    console.log( { entity },{ userId });
+
+    let user;
+
+    if (userId) {
+      //find user detail
+      user = await this.userRepository.findOne({ where: { id: userId } });
+
+      if (has(entity, 'email') && user?.email !== entity.email) {
+        const exists = await this.checkIfExistsEmail(entity.email);
```

Comment:
How did you mange if passing same email in api, I mean not make changes, it should not throw error.

Reply:
@kishanKhandala  Still this is pending